src/nvim/option.c

```diff
@@ -3984,235 +3984,253 @@ set_num_option (
     return e_secure;
   }
 
-  *pp = value;
-  /* Remember where the option was set. */
-  set_option_scriptID_idx(opt_idx, opt_flags, current_SID);
-
-  if (curbuf->b_p_sw < 0) {
-    errmsg = e_positive;
-    curbuf->b_p_sw = curbuf->b_p_ts;
+  // Many number options assume their value is in the signed int range.
+  if (value < INT_MIN || value > INT_MAX) {
+      return e_invarg;
   }
 
-  /*
-   * Number options that need some action when changed
-   */
-  if (pp == &p_wh || pp == &p_hh) {
-    if (p_wh < 1) {
+  // Options that need some validation.
+  if (pp == &p_wh) {
+    if (value < 1) {
       errmsg = e_positive;
-      p_wh = 1;
-    }
-    if (p_wmh > p_wh) {
+    } else if (p_wmh > value) {
       errmsg = e_winheight;
-      p_wh = p_wmh;
     }
-    if (p_hh < 0) {
+  } else if (pp == &p_hh) {
+    if (value < 0) {
       errmsg = e_positive;
-      p_hh = 0;
-    }
-
-    /* Change window height NOW */
-    if (lastwin != firstwin) {
-      if (pp == &p_wh && curwin->w_height < p_wh)
-        win_setheight((int)p_wh);
-      if (pp == &p_hh && curbuf->b_help && curwin->w_height < p_hh)
-        win_setheight((int)p_hh);
     }
-  }
-  /* 'winminheight' */
-  else if (pp == &p_wmh) {
-    if (p_wmh < 0) {
+  } else if (pp == &p_wmh) {
+    if (value < 0) {
       errmsg = e_positive;
-      p_wmh = 0;
-    }
-    if (p_wmh > p_wh) {
+    } else if (value > p_wh) {
       errmsg = e_winheight;
-      p_wmh = p_wh;
     }
-    win_setminheight();
   } else if (pp == &p_wiw) {
-    if (p_wiw < 1) {
+    if (value < 1) {
       errmsg = e_positive;
-      p_wiw = 1;
+    } else if (p_wmw > value) {
+      errmsg = e_winwidth;
     }
-    if (p_wmw > p_wiw) {
+  } else if (pp == &p_wmw) {
+    if (value < 0) {
+      errmsg = e_positive;
+    } else if (value > p_wiw) {
       errmsg = e_winwidth;
-      p_wiw = p_wmw;
     }
-
-    /* Change window width NOW */
-    if (lastwin != firstwin && curwin->w_width < p_wiw)
-      win_setwidth((int)p_wiw);
-  }
-  /* 'winminwidth' */
-  else if (pp == &p_wmw) {
-    if (p_wmw < 0) {
+  } else if (pp == &p_mco) {
+    if (value > MAX_MCO) {
+      errmsg = e_invarg;
+    } else if (value < 0) {
       errmsg = e_positive;
-      p_wmw = 0;
     }
-    if (p_wmw > p_wiw) {
-      errmsg = e_winwidth;
-      p_wmw = p_wiw;
+  } else if (pp == &p_titlelen) {
+    if (value < 0) {
+      errmsg = e_positive;
     }
-    win_setminheight();
```

Comment:
this is the only side-effect that got removed. `win_setminheight` doesn't seem to have anything to do with `winminwidth`